src/order/locally_finite.lean

```diff
@@ -646,3 +648,44 @@ begin
 end
 
 end finset
+
+namespace finset
+variables {γ : Type*} [linear_order γ] [locally_finite_order γ]
+  [decidable (nonempty γ)] [is_well_order γ (<)]
+
+/--
+The finset of elements `x` such that `x ≤ a`. Basically `set.Iic a` as a finset.
+
+In contrast to `finset.Iic`, this definition does not require the type to be inhabited, but it does
+require `linear_order` and `is_well_order` instead. -/
+noncomputable def Iic_wf (a : γ) : finset γ :=
+if h : nonempty γ
```

Comment:
You can provide a proof of `nonempty γ` directly via `⟨a⟩`.

Reply:
Right, thanks!